pkgs/os-specific/linux/amneziawg/default.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  kernel,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "amneziawg";
+  version = "1.0.20240711";
+
+  src = fetchFromGitHub {
+    owner = "amnezia-vpn";
+    repo = "amneziawg-linux-kernel-module";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-WOcBTxetVz2Sr62c+2aGNyohG2ydi+R+az+4qHbKprI=";
+  };
+
+  sourceRoot = "${finalAttrs.src.name}/src";
+  hardeningDisable = [ "pic" ];
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  buildFlags = [
+    "apply-patches"
+    "module"
+  ];
+
+  makeFlags =
+    kernel.makeFlags
+    ++ [ "KERNELDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build" ]
+    ++ lib.optional (lib.versionAtLeast kernel.version "5.6") "KERNEL_SOURCE_DIR=${kernel.src}";
```

Comment:
`kernel.src` does not work on my system.

`kernel.src` is `«derivation /nix/store/1ydcb00j8j88s97q9dlbcpwzack0ijg6-linux-6.6.47.tar.xz.drv»`

I was able to fix it by using
```nix
stdenv.mkDerivation {
  inherit (kernel) src version;
  pname = "kernel-source";

  phases = [
    "unpackPhase"
    "installPhase"
  ];

  installPhase = ''
    mkdir $out
    mv * $out
  '';
}
```
instead of `kernel.src`. It probably isn't the most correct and/or clean solution tho.

Reply:
What do you mean as "doesn't work"? Can you, please, show build logs and kernel-related nixos configuration?

Reply:
```
error: builder for '/nix/store/ymg3ja2y99bnzvcy7m08c741anm25zpf-amneziawg-1.0.20240711.drv' failed with exit code 2;
       last 18 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/yi3d68fv07hf03c4fjlx78dx0mz9wxb1-source
       > source root is source/src
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > build flags: -j8 SHELL=/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash O=\$\(buildRoot\) CC=/nix/store/lbk30k56awz9vz9qpid93fkjns0xwlhd-gcc-wrapper-13.3.0/bin/cc HOSTCC=/nix/store/lbk30k56awz9vz9qpid93fkjns0xwlhd-gcc-wrapper-13.3.0/bin/cc HOSTLD=/nix/store/qrw9mznq4p1135k53aa5g9saz229srf4-binutils-wrapper-2.42/bin/ld ARCH=x86_64 KERNELDIR=/nix/store/chalmld7wyb6r5n4cya0z5cyv509kfrg-linux-6.6.47-dev/lib/modules/6.6.47/build KERNEL_SOURCE_DIR=/nix/store/pb1d0qsnzxv8larpas16nbk5gfhdzpvf-linux-6.6.47.tar.xz apply-patches module
       > Makefile:80: ***
       > You're running a modern Linux Kernel (version 6.6.47).
       >
       > In order to build AmneziaWG kernel module for this kernel you must obtain sources of your kernel
       > by yourself and make a symlink to them into this directory:
       >
       >     ln -s <path to kernel sources> kernel
       >
       > After that please run make script again.  Stop.
       For full logs, run 'nix log /nix/store/ymg3ja2y99bnzvcy7m08c741anm25zpf-amneziawg-1.0.20240711.drv'.
```

it fails due to `$(KERNEL_SOURCE_DIR)` being an archive instead of directory

in (almost) minimal possible setup it still fails. i am currently trying to minimize it further to put it on gist

Reply:
found minimal example: https://gist.github.com/nikitax44/52f5284fa146e52b7aca252fb6d81564

Reply:
I can reproduce it with normal kernels but not zen kernel I use on my desktop. Fixed by `kernel.src` -> `srcOnly kernel`.

Reply:
I also think `ARCH` is not needed because it is set by `kernel.makeFlags`. I didn't checked `CROSS_COMPILE`, but, I think, it should be set too.

Use `nix --extra-experimental-features repl-flake repl ".#nixosConfigurations.hostname"`, `config.boot.kernelPackages.amneziawg.makeFlags` if you want to check current `makeFlags`.

Reply:
@nikitax44 are your issue resolved?

Reply:
Sorry, didn't see the notification. Yes, it was resolved.